src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,567 @@
+
+            # Traversing thorugh all agents
+            for each_agent_pod in arc_agents_pod_list.items:
+
+                # Fethcing the current Pod name and creating a folder with that name inside the timestamp folder
+                agent_name = each_agent_pod.metadata.name
+                arc_agent_logs_path = os.path.join(filepath_with_timestamp, "Arc_Agents_logs")
+                try:
+                    os.mkdir(arc_agent_logs_path)
+                except FileExistsError:
+                    pass
+
+                agent_name_logs_path = os.path.join(arc_agent_logs_path, agent_name)
+                try:
+                    os.mkdir(agent_name_logs_path)
+                except FileExistsError:
+                    pass
+
+                # If the agent is not in Running state we wont be able to get logs of the containers
+                if(each_agent_pod.status.phase != "Running"):
+                    continue
+
+                # Traversing through all of the containers present inside each pods
+                for each_container in each_agent_pod.spec.containers:
+
+                    # Fetching the Container name
+                    container_name = each_container.name
+
+                    # Creating a text file with the name of the container and adding that containers logs in it
+                    container_log = corev1_api_instance.read_namespaced_pod_log(name=agent_name, container=container_name, namespace="azure-arc")
+
+                    arc_agent_container_logs = os.path.join(agent_name_logs_path, container_name + ".txt")
+                    with open(arc_agent_container_logs, 'w+') as container_file:
+                        container_file.write(str(container_log))
+
+    except OSError as e:
+        storage_space_available = False
+        shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+
+    except Exception as e:
+        logger.warning("An expection has occured while trying to fetch the azure arc agents logs from the cluster. Exception: {}".format(str(e)) + "\n")
+    
+    return storage_space_available
+
+
+def arc_agents_event_logger(filepath_with_timestamp, storage_space_available):
+
+    try:
+
+        if storage_space_available :
+
+            # CMD command to get helm values in azure arc and converting it to json format
+            command = ["kubectl", "get", "events", "-n", "azure-arc", "--output", "json"]
+
+            # Using subprocess to execute the helm get values command and fetching the output
+            resposne_kubectl_get_events = Popen(command, stdout=PIPE, stderr=PIPE)
+            output_kubectl_get_events, error_kubectl_get_events = resposne_kubectl_get_events.communicate()
+
+            if resposne_kubectl_get_events.returncode != 0:
+                if ('forbidden' in error_kubectl_get_events.decode("ascii") or 'timed out waiting for the condition' in error_kubectl_get_events.decode("ascii")):
```

Comment:
this logic was specific for helm commands. does this apply for kubectl commands as well?